### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by beyondbill**
Add resource specs and examples to kick off discussion over the design.

As for implementation (which is **not** included in this PR), the current design should be easy with the help of [github.com/araddon/dateparse](https://github.com/araddon/dateparse), which can automatically detect the layout of a timestamp string and parse it.

Other things I want to discuss:
- Where to publish this? I think this provider can be published publicly in [Terraform Registry](https://registry.terraform.io/) under takescoop org for free. Do we have an org there yet? We may want to create a project to manage Terraform Registry in Terraform in mars
- Any suggestion on the license?

TODO list after this PR:
- Implementation
- Testing
- CI (lint, build, test and publish)
- Readme
- Licensing

### CODE REVIEW

**Comment by bendrucker** [comment]
> Where to publish this? I think this provider can be published publicly in Terraform Registry under takescoop org for free.

Yes, absolutely. We can set up that app for our GitHub org. The registry pulls releases from GitHub releases.

https://www.terraform.io/docs/registry/providers/publishing.html

HashiCorp provides official GitHub Actions. You don't have to use them, but reproducing this in Circle will be a pain. So either we'd need to allow GitHub Actions in our org or you'd have to spend a day or two figuring out how to port things to Circle config and get a GH token added. I'd recommend pursuing the Actions path.

> Any suggestion on the license?

MIT, by default

**Review by bendrucker — CHANGES_REQUESTED** [review]
Rather than have the examples in a module like, can you write some docs?

https://www.terraform.io/docs/registry/providers/docs.html

I have lots of comments to add around behavioral settings for different fields. The schema looks reasonable at first glance. But I'd rather see the functionality described in plain English first, and then we can work backwards into the resource schema details.

**Comment by bendrucker** [comment]
I'll save Go/TF Plugin SDK comments until we have the API documented and give those to you in one big batch.

**Comment by beyondbill** [comment]
@bendrucker I have finished the TF docs and moved the examples there. Want to take a look before I go into implementation?

**Comment by bendrucker** [comment]
On it!

File: docs/data-sources/timestamp.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# 2020-11-22T11:11:11Z08:00
+data "localtime_timestamp" "custom_layout" {
+  local_time = "2020/11/22 11:11:11"
+  layout     = "2006-01-02T15:04:05Z07:00"
+}
+```
+
+## Argument Reference
+
+- `local_time` - (Required) The local timestamp without timezone info. It can be in any valid layout.
```
Comment (bendrucker) [inline docs/data-sources/timestamp.md]: "Any valid layout" is ambiguous/circular. Give some examples.

File: docs/data-sources/timestamp.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+- `location` - (Optional) The location of the timezone where the exported timestamp will be. It must be a valid [tz database name](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List). Default to `Local`.
```
Comment (bendrucker) [inline docs/data-sources/timestamp.md]: "Local" is not part of the TZ database, it's a [`LoadLocation`](https://golang.org/pkg/time/#example_FixedZone) behavior. Probably best not to expose that and supply "Local" internally if this field is not set.

File: docs/data-sources/timestamp.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- `layout` - (Optional) The layout that the exported timestamp will use. It must follow the [Golang format](https://golang.org/pkg/time/#pkg-constants). If this parameter is provided, `timezone_layout` will be ignored. Otherwise the exported timestamp will use the same layout as in `local_time`, plus `timezone_layout`.
```
Comment (bendrucker) [inline docs/data-sources/timestamp.md]:
```suggestion
- `layout` - (Optional) The layout that the exported timestamp will use. It must follow the [Golang format](https://golang.org/pkg/time/#pkg-constants). This attribute conflicts with `timezone_layout`.
```

File: docs/data-sources/timestamp.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- `timezone_layout` - (Optional) The timezone part of `layout`. It's useless when `layout` is provided. Default to " -0700".
```
Comment (bendrucker) [inline docs/data-sources/timestamp.md]: Why is this a separate attribute at all? Why can't the time zone be inserted into the layout?

Comment (bendrucker) [reply docs/data-sources/timestamp.md]: Also, -0700 is an arbitrary default. If this field needs to exist at all, it should be user supplied with no default

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Localtime Provider
```
Comment (bendrucker) [inline docs/index.md]:
```suggestion
# Local Time Provider
```

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+
+The localtime provider converts a local timestamp (e.g. without timezone info) into one with the specified timezone and layout.
```
Comment (bendrucker) [inline docs/index.md]:
```suggestion
The localtime provides resources for dealing with local times and time zones.
```

File: docs/data-sources/timestamp.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# Data Source `timestamp`
+
+The timestamp data source allows you to converts a local timestamp (e.g. without timezone info) into one with the specified timezone and layout.
+
+## Example Usages
+
+### Use local timezone
```
Comment (bendrucker) [inline docs/data-sources/timestamp.md:7]: I'd trim down this examples to a single canonical one, _maybe_ one simple and one advanced. If a user wants to understand how to use uncommon optional attributes they can look at the attribute reference.

**Review by bendrucker — COMMENTED** [review]
Would recommend keeping just `docs/` in this PR and saving the code for a separate one. Still have plenty of comments there.

Will be keeping an eye out for review requests here, feel free to send em my way as fast as you can!

File: docs/data-sources/timestamp.md (hunk 3 above)
Comment (beyondbill) [reply docs/data-sources/timestamp.md]: @bendrucker Do you want the behavior to be a conflict rather than ignore?

File: docs/data-sources/timestamp.md (hunk 4 above)
Comment (beyondbill) [reply docs/data-sources/timestamp.md]: @bendrucker The idea is to provide an easy way to keep the existing layout of local time and add a timezone. You'll have a better idea after seeing the example in the doc. It has to have a default since local_time does not provide a layout for the timezone part

File: docs/data-sources/timestamp.md (hunk 3 above)
Comment (bendrucker) [reply docs/data-sources/timestamp.md]: Yes, attributes that are populated should never be ignored. They should either be consumed or rejected with a validation error.

File: docs/data-sources/timestamp.md (hunk 4 above)
Comment (bendrucker) [reply docs/data-sources/timestamp.md]: Ok, I think I get it. The implementation is leaking.

A few things come to mind:

* Trim the leading space
* Call it `layout_timezone`
* Clarify that this is a layout, and not a literal time zone value. Give examples of equivalent layouts to illustrate this point, e.g., `-0700` and `MST`.
* Use formal schema language (e.g. "conflicts") not informal statements like "it's useless." And explain how it conflicts with the other field (layout), i.e. that this value should be included as part of `layout` if specified

File: docs/data-sources/timestamp.md (hunk 3 above)
Comment (bendrucker) [reply docs/data-sources/timestamp.md]: Also, avoid describing behavior when the field isn't set if there's a conflicting field. Put that description in the conflicting field.

**Review by bendrucker — COMMENTED** [review]
Nice, few minor suggestions re: `timezone_layout` / `layout_timezone` for clarity. Seems like a workable schema though!

**Comment by beyondbill** [comment]
@bendrucker Great feedback on `layout_timezone`! I'll make changes later this afternoon

**Comment by bendrucker** [comment]
Cool! Otherwise looks good, I'll come back with a ✅ whenever you commit those changes. And then if you wanna open up a draft for the implementation ASAP, I can start dropping comments in roughly this order:

1. Schema (particularly behavioral fields)
2. Behavior (i.e. the `Read` method)
3. Tests

TDD is certainly helpful, but if you wanna bite off a smaller chunk I can review the schema early on.

**Comment by beyondbill** [comment]
@bendrucker This should be good to go! Next PR will focus on schema code. Another one for behaviors+tests

**Review by bendrucker — APPROVED** [review]
